code/game/objects/items/reagent_containers/jerrycan.dm

```diff
@@ -0,0 +1,45 @@
+/obj/item/reagent_containers/jerrycan
+	name = "\improper jerry can"
+	desc = "A can filled with fuel to light things on fire. It has Absolut Jerry stamped in the side."
+	icon = 'icons/obj/items/tank.dmi'
+	icon_state = "canister"
+	volume = 200
+	list_reagents = list(/datum/reagent/fuel = 200)
+	///how much fuel we use up per spill
+	var/fuel_usage = 10
+
+/obj/item/reagent_containers/jerrycan/afterattack(atom/A, mob/user)
+	if(!(get_dist(user,A) <= 1))
+		return
+	if(isturf(A))
```

Comment:
If you could add an `attack_turf()` proc call to `/turf/attackby()`, and place this inside of it for your object, that'd be ideal